var/spack/repos/builtin/packages/mesa/package.py

```diff
@@ -34,65 +39,105 @@ class Mesa(AutotoolsPackage):
     version('13.0.6', '1e5a769bc6cfd839cf3febcb179c27cc')
     version('12.0.6', '1a3d4fea0656c208db59289e4ed33b3f')
     version('12.0.3', '1113699c714042d8c4df4766be8c57d8')
 
-    variant('swrender', default=False,
-        description="Build with (gallium) software rendering.")
-
+    variant('swrender', default=True,
+            description="Build with (gallium) software rendering.")
+    variant('hwrender', default=False,
+            description="Build with (DRI) hardware rendering.")
     variant('llvm', default=False,
-        description="Use llvm for rendering pipes.")
+            description="Use llvm for rendering pipes.")
 
     # General dependencies
-    depends_on('python@2.6.4:')
-    depends_on('py-mako@0.3.4:', type=('build', 'run'))
+    depends_on('pkg-config@0.9.0:', type='build')
     depends_on('flex@2.5.35:', type='build')
     depends_on('bison@2.4.1:', type='build')
-    depends_on('pkg-config@0.9.0:', type='build')
-
-    # Off-screen with llvmpipe
-    # Note: there must be a better way of selecting the preferred llvm
-    depends_on('llvm+link_dylib', when='+llvm')
-
-    # For DRI and hardware acceleration
+    depends_on('binutils', type='build')
+    depends_on('python@2.6.4:', type='build')
+    depends_on('py-mako@0.3.4:', type='build')
+    depends_on('gettext')
+    depends_on('icu4c')
+    depends_on('expat')
     depends_on('libpthread-stubs')
-    depends_on('libdrm')
     depends_on('openssl')
+    depends_on('xproto')
+    depends_on('glproto@1.4.14:')
+    depends_on('presentproto@1.0:')
     depends_on('libxcb@1.9.3:')
-    depends_on('libxshmfence@1.1:', when='~swrender')
-    depends_on('libx11', when='~swrender')
-    depends_on('libxext', when='~swrender')
-    depends_on('libxdamage', when='~swrender')
+    depends_on('libx11')
+    depends_on('libxext')
+    depends_on('libxshmfence@1.1:')
+    depends_on('libxdamage')
     depends_on('libxfixes')
-    # depends_on('expat', when='~swrender')
-    depends_on('libelf', when='+llvm~swrender')
+    depends_on('libxv')
+    depends_on('libxvmc')
 
-    depends_on('glproto@1.4.14:', type='build', when='~swrender')
-    depends_on('dri2proto@2.6:', type='build', when='~swrender')
-    depends_on('dri3proto@1.0:', type='build', when='~swrender')
-    depends_on('presentproto@1.0:', type='build', when='~swrender')
+    # For DRI and hardware acceleration
+    depends_on('dri2proto@2.6:', type='build')
+    depends_on('dri3proto@1.0:', type='build')
+    depends_on('libdrm')
 
-    # TODO: Add package for systemd, provides libudev
-    # Using the system package manager to install systemd didn't work for me
+    depends_on('llvm@:3.8.1+link_dylib', when='@12:12.99+llvm')
+    depends_on('llvm@:3.9.1+link_dylib', when='@13:13.99+llvm')
+    depends_on('llvm+link_dylib', when='+llvm')
+    depends_on('libelf', when='+llvm')
 
     def configure_args(self):
+        """Build drivers for platforms supported by spack;
```

Comment:
The prior default is ```~swrender```. If one now specifies ```~swrender~hwrender``` the following args are added to configure relative to the prior behavior:

* ```--enable-glx```
* ```--enable-glx-tls```
* ```--disable-xa```
* ```--disable-dri3```
* ```--disable-gallium-osmesa```
* ```--enable-osmesa```

The following are removed:

* ```--disable-gles1```
* ```--disable-glx```
* ```--enable-texture-float```
* ```--enable-gallium-osmesa```

(I'm just listing this out here for reference so folks who are more familiar can glance at this and see if they have an opinion)

Reply:
I wouldn't place too much weight on the current version of mesa. #4602 was merged while we were still working on it. This new version may not be the last word on the subject, but it should satisfy a range of purposes.